scylla/src/transport/cluster.rs

```diff
@@ -384,8 +358,6 @@ impl ClusterData {
             )
         }
 
-        Self::update_rack_count(&mut datacenters);
-
         let keyspaces = metadata.keyspaces;
```

Comment:
...was that really pointless?! I can't believe it has always been... Could you please locate the commit that made `datacenters` unused?

Reply:
This code was introduced in this commit: https://github.com/scylladb/scylla-rust-driver/commit/8d82b043e1bc154330de78c183ea1d311ce4b2b9#diff-fc8ab4fd7742b17e4767ade6844fd697e5a7b0050bbadb414f52d19f6245bbac (PR: https://github.com/scylladb/scylla-rust-driver/pull/172 )
You can see that the result is saved in ClusterData.

Then the field in ClusterData was removed by the same author 2 years later: https://github.com/scylladb/scylla-rust-driver/commit/0db57c160924330cd1d92c418887e4feb4d4fd03 - but the calculation in ClusterData::new remained, introducing uselessness. PR: https://github.com/scylladb/scylla-rust-driver/pull/612

Reply:
I'm pretty sure that even thought there was no useless calculation before #612 this API was still useless because `Datacenter` struct was never truly public (= pub and exported)